client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -178,15 +179,32 @@ public ClientFactoryImpl(String scope, Controller controller, ConnectionFactory
         Consumer<Segment> segmentSealedCallBack = s -> {
             throw new IllegalStateException("RevisionedClient: Segmentsealed exception observed for segment:" + s);
         };
-        String delegationToken = Futures.getAndHandleExceptions(controller.getOrRefreshDelegationTokenFor(segment.getScope(),
-                segment.getStreamName()), RuntimeException::new);
+
+        String delegationToken = checkStreamReturnDelegation(streamName, segment);
+
         SegmentOutputStream out = outFactory.createOutputStreamForSegment(segment, segmentSealedCallBack,
                 config.getEventWriterConfig(), delegationToken);
         ConditionalOutputStream cond = condFactory.createConditionalOutputStream(segment, delegationToken, config.getEventWriterConfig());
         SegmentMetadataClient meta = metaFactory.createSegmentMetadataClient(segment, delegationToken);
         return new RevisionedStreamClientImpl<>(segment, in, out, cond, meta, serializer);
     }
 
+    private String checkStreamReturnDelegation(String streamName, Segment segment) {
```

Comment:
It's not obvious from the name that this throws in the event the segment is sealed. Please add a Javadoc and include this information.

Reply:
javadoc added